syntax = "proto3";

package product;

// ðŸ‘‡ This tells protoc where to place generated Go code
option go_package = "product/proto/productpb";

message Product {
    string id = 1;
    string name = 2;
    double price = 3;
}

message GetProductByIDRequest {
    string id = 1;
}

message GetProductResponse {
    Product product = 1;
}


message GetProductsRequest { 
    // Both fields are optional for filtering
    string name = 1;
    float price = 2;
}


message GetProductsResponse {
    repeated GetProductResponse products = 1;
}

service ProductService {
    rpc GetProductByID (GetProductByIDRequest) returns (GetProductResponse);
    rpc GetProducts (GetProductsRequest) returns (GetProductsResponse);
}


